core/src/Cuda/Kokkos_Cuda_Graph_Impl.hpp

```diff
@@ -121,10 +121,11 @@ struct GraphImpl<Kokkos::Cuda> {
   }
 
   template <class NodeImpl>
-  //  requires NodeImplPtr is a shared_ptr to specialization of GraphNodeImpl
-  //  Also requires that the kernel has the graph node tag in its policy
-  void add_node(std::shared_ptr<NodeImpl> const& arg_node_ptr) {
-    static_assert(NodeImpl::kernel_type::Policy::is_graph_kernel::value);
+  std::enable_if_t<
+      Kokkos::Impl::is_graph_kernel_v<typename NodeImpl::kernel_type>>
```

Comment:
I am confused.  Where is that `Impl::is_graph_kernel_v` helper variable template defined?

Reply:
Not easy to find indeed.

https://github.com/kokkos/kokkos/blob/bc6581798c5efe1abc8b1b937baa7f2185414665/core/src/Kokkos_Concepts.hpp#L168